### PR TITLE
Feature - Delete job and talent post

### PR DESCRIPTION
**Pull request by jacobtyq**
Description
---
- Allow only job post creator to delete a post, and reload page once deleted
- Handle errors with an appropriate error message

Possible To Do
---
- Convert Modal to a reusable component

Gif
---
Successful Deletion of Job / Talent Post
delete-post

The menu is hidden if you are not Job / Talent Post creator
Screenshot 2020-09-27 at 4 00 07 PM

Error 
Screenshot 2020-09-27 at 3 59 08 PM

### CODE REVIEW

File: ui/src/shared/components/PostDetail.tsx [hunk 1]
```diff
@@ -151,6 +161,7 @@ const PostDetail: React.FC<PostDetailProps> = function (props) {
           Connect with {user.first_name}
         </Button>
       </ButtonContainer>
+      <Modal></Modal>
```
Comment (SheldonCheng) [inline ui/src/shared/components/PostDetail.tsx:163]: Did you mean to remove this line?

File: ui/src/shared/components/PostDetail.tsx [hunk 2]
```diff
@@ -127,18 +134,21 @@ const PostDetail: React.FC<PostDetailProps> = function (props) {
...
-                    <EditIcon />
-                    Edit
-                  </StyledMenuItem>
-                  <StyledMenuItem>
-                    <DeleteIcon />
-                    <DangerText>Delete Post</DangerText>
-                  </StyledMenuItem>
-                </StyledMenuList>
-              </Menu>
+              'Context:' { context?.profile?.id }  'User:' { user.id } | 'Data:' { category }
```
Comment (SheldonCheng) [inline ui/src/shared/components/PostDetail.tsx]: image

Eh not sure what is going on here

Comment (jacobtyq) [reply ui/src/shared/components/PostDetail.tsx]: Oops meant for debugging purposes, have removed it

File: ui/src/shared/components/PostDetail.tsx (hunk 1 above)
Comment (jacobtyq) [reply ui/src/shared/components/PostDetail.tsx:163]: Removed

**Comment by SheldonCheng** [comment]
@jacobtyq The app crashes if I try to delete a post now. I've tried to debug and it seems that deleting the `<Modal></Modal>` line broke it. This seems like a code smell to me. In any case can restore that line and get it merged in first.

**Comment by jacobtyq** [comment]
> @jacobtyq The app crashes if I try to delete a post now. I've tried to debug and it seems that deleting the `<Modal></Modal>` line broke it. This seems like a code smell to me. In any case can restore that line and get it merged in first.

Yup ok placed it back

**Comment by jacobtyq** [comment]
@SheldonCheng for your review again 🙇

**Review by SheldonCheng — APPROVED** [review]

**Pull request merged**